service/k8sclient/k8client.go

```diff
@@ -94,10 +93,11 @@ type K8Client struct {
 }
 
 // NewK8Client returns new K8Client object.
-func NewK8Client(logger logger.Logger) *K8Client {
+func NewK8Client(ctx context.Context) (*K8Client, error) {
+	kubeCtl, err := kubectl.NewKubeCtl(ctx)
 	return &K8Client{
-		kubeCtl: kubectl.NewKubeCtl(logger),
-	}
+		kubeCtl: kubeCtl,
+	}, err
```

Comment:
I think we should use slightly more traditional error handling there.
![image](https://user-images.githubusercontent.com/11512/89643190-e9bb3480-d8bd-11ea-9d79-5ad97d125da1.png)